File: 02-LINKED-LISTS/reverse-linked-list.php
<?php

# https://leetcode.com/problems/reverse-linked-list/
class Solution {
    function reverseList(?ListNode $head): ?ListNode
    {
        $prev = null;
        $cur = $head;

        while ($cur !== null) {
            $next_node = $cur->next;
            $cur->next = $prev;
            $prev = $cur;
            $cur = $next_node;
        }

        return $prev;
    }
}

File: 02-LINKED-LISTS/Python/reverse-linked-list.py
from typing import *
import ListNode

# https://leetcode.com/problems/reverse-linked-list/
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        cur = head

        while cur:
            next_node = cur.next
            cur.next = prev
            prev = cur
            cur = next_node

        return prev

